src/cascadia/TerminalCore/TerminalDispatch.cpp

```diff
@@ -134,6 +134,69 @@ try
 }
 CATCH_LOG_RETURN_FALSE()
 
+bool TerminalDispatch::HorizontalTabSet() noexcept
+{
+    return true;
+}
+
+bool TerminalDispatch::ForwardTab(const size_t numTabs) noexcept
+{
+    const auto cursorPosition = _terminalApi.GetCursorPosition();
+    auto column = cursorPosition.X;
+    const auto row = cursorPosition.Y;
+    auto tabsPerformed = 0u;
+    const auto width = 80;
+    _InitTabStopsForWidth(width);
+    while (column + 1 < width && tabsPerformed < numTabs)
+    {
+        column++;
+        if (til::at(_tabStopColumns, column))
+        {
+            tabsPerformed++;
+        }
+    }
+
+    return _terminalApi.SetCursorPosition(column, row);
+}
+
+bool TerminalDispatch::BackwardsTab(const size_t numTabs) noexcept
+{
+    const auto cursorPosition = _terminalApi.GetCursorPosition();
+    auto column = cursorPosition.X;
+    const auto row = cursorPosition.Y;
+    auto tabsPerformed = 0u;
+    const auto width = 80;
+    _InitTabStopsForWidth(width);
+    while (column > 0 && tabsPerformed < numTabs)
+    {
+        column--;
+        if (til::at(_tabStopColumns, column))
+        {
+            tabsPerformed++;
+        }
+    }

```

Comment:
This definitely is out of scope for this PR, but I wonder if there isn't some way that we could have a common implementation for both `AdaptDispatch` and `TerminalDispatch` for these methods. Since the `_tabStops` are stored internal to the dispatcher, it seems like both of them could somehow have a common implementation...

Reply:
FYI, my long term plan is to replace the `TerminalDispatch` class entirely with `AdaptDispatch`, so the common implementation in this case will ultimately just be `AdaptDispatch`. For now, though, I've been trying to keep the `TerminalDispatch` implementation as close as possible to the corresponding `AdaptDispatch` code, just with `_pConApi` changed to `_terminalApi`. I'm hoping that'll make things easier to synchronize when we eventually merge the two.

Reply:
@j4james your plan makes sense. Would you still like me to continue the work here? Or can you point me the correct direction towards merging these two?

Reply:
There's a ton of work that needs to be done on the conapi interface before we can merge. Other than that, we need to make sure that everything that's currently only implemented in `TerminalDispatch` is also copied across to `AdaptDispatch`. And as I said above, if you're copying functionality from `AdaptDispatch` (like these tabs), it's preferable if the code is essentially identical between the two.

So as long as you're doing that, I don't think there's anything wrong with this PR. I'm just not sure why we actually need the functionality in `TerminalDispatch` at this point. Are we intending to pass these tab sequences through conpty?

Reply:
Oh, I like that plan. That would be nice. I knew this would be out-of-scope for this PR, but this is a good vision to have.

I think the bigger vision here is @skyline75489 is working on tmux control mode, and that needs much more complete VT support than `TerminalDispatch` offers right now. Eventually we'll want it for passthrough mode, but that's not really in the cards right now.